src/FluidHydratorFactory.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace TheCodingMachine\FluidHydrator;
+
+use MetaHydrator\Database\DBProvider;
+
+class FluidHydratorFactory
```

Comment:
I'm not really comfortable with the notion of the `FluidHydratorFactory` here.
You are passing the factory to the `FluidHydrator`... the FluidHydrator depends on the factory that created it which is really unusual... The factory acts like a services repository and it's not its role (a factory should not have a getter).

Wouldn't it be easier to simply pass the DBProvider as an optionnal parameter of the `FluidHydrator` constuctor (and completely get rid of the `FluidHydratorFactory`?